<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jes" />


<title>Introduction to ggplot2 and dplyr</title>

<script src="site_libs/header-attrs-2.19/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>







<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="computer-setup.html">Computer Setup</a>
</li>
<li>
  <a href="NEON-data-intro.html">NEON Data</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Modules
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="00-module-contents.html">Overview</a>
    </li>
    <li>
      <a href="01-A-R-intro.html">Intro to R</a>
    </li>
    <li>
      <a href="01-B-Rmarkdown-intro.html">R markdown</a>
    </li>
    <li>
      <a href="01-C-R-workshop.html">R workshop</a>
    </li>
    <li>
      <a href="02-A-tidyr.html">tidyr</a>
    </li>
    <li>
      <a href="02-B-git-intro.html">Intro to git</a>
    </li>
  </ul>
</li>
<li>
  <a href="exercises-guide.html">Exercises</a>
</li>
<li>
  <a href="download-student-RStudio-project.html">RStudio Workbook</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Introduction to ggplot2 and dplyr</h1>
<h4 class="author">Jes</h4>

</div>


<p><strong>Assigned Reading:</strong></p>
<blockquote>
<p>Chapters 1.1, 1.2, 2, 3, 9 and 10 in: Wickham, H. 2009. <em>ggplot2:
Elegant Graphics for Data Analysis.</em> Springer. DOI: <a
href="https://link.springer.com/book/10.1007%2F978-0-387-98141-3">Stanford
Full Text</a></p>
<p>The sections on mutating and filtering joins in <a
href="https://cran.r-project.org/web/packages/dplyr/vignettes/two-table.html">this
vignette</a></p>
</blockquote>
<p><strong>Optional Reading:</strong></p>
<blockquote>
<p>Chapter 4 in: Wickham, H. 2009. <em>ggplot2: Elegant Graphics for
Data Analysis.</em> Springer. DOI: <a
href="https://link.springer.com/book/10.1007%2F978-0-387-98141-3">Stanford
Full Text</a></p>
</blockquote>
<p>RStudio has produced several helpful <a
href="https://www.rstudio.com/resources/cheatsheets/">cheatsheets</a> on
data management and visualization with ggplot2 and tidyr. You may want
to view or print one or several of them.</p>
<div id="before-class" class="section level3">
<h3>Before Class</h3>
<p>Download <a
href="https://github.com/FukamiLab/BIO202/blob/master/code/02-A-tidyr-exercise-code.R">this
R script</a> and save it in your code folder in the research project you
created on the first day of class. You will need to click the ‘Raw’
button and then download the text file that appears. Be sure to remove
the .txt extension that may have been added when you saved the file to
your computer so that the file ends in .R. You may want to run the first
part of the script that downloads the data to be sure that there will be
no issues with data access during class.</p>
</div>
<div id="key-points" class="section level3">
<h3>Key Points</h3>
<ul>
<li>Data should be organized so that observations are in rows and
attributes (variables) are in columns.</li>
<li>The <code>dplyr</code> package contains functions for manipulating
data:
<ul>
<li><code>spread</code> and <code>gather</code> change the format of
data tables.</li>
<li><code>separate</code> and <code>unite</code> change the format of
variables.</li>
<li><code>filter</code> creates a subset of a data table.</li>
<li><code>mutate</code> creates new variables.</li>
<li><code>group_by</code> and <code>summarise</code> are used to
summarize observations according to the values of their variables.</li>
</ul></li>
<li>The pipe function <code>%&gt;%</code> is used to link several
operations together (e.g. function composition).</li>
<li>The <code>ggplot2</code> package is set of functions for visualizing
data.
<ul>
<li>A ggplot2 plot contains three components: (1) the data, (2) the
aesthetic mappings between variables and visual properties, and (3)
layers describing how to display the observations.</li>
<li>The aesthetic mapping <code>aes</code> must minimally describe which
variables define the plot coordinate space.</li>
<li><code>facet_wrap</code> creates plots for specified data
subsets.</li>
</ul></li>
</ul>
</div>
<div id="manipulating-data-with-dplyr" class="section level3">
<h3>Manipulating data with dplyr</h3>
<p>Open the R project that you created on the first day of class. Then
when R Studio opens, open the R script file that you downloaded for
today’s class (see above). This script downloads a data set from our
course website and then proceeds to manipulate and summarise the data.
These data contain three tables from the Winter 2016 BIO46 class on
lichen microorganisms:</p>
<ol style="list-style-type: decimal">
<li><strong>trees</strong> is a table where observations are trees on
which lichens were collected. Columns describe each tree’s location,
elevation, and species name.</li>
<li><strong>lichens</strong> is a table where each observation is a
lichen that was collected. Columns describe the tree on which the lichen
was collected, the team that collected the lichen, the collection date,
and the lichen species name.</li>
<li><strong>algae</strong> is a table where each observation is a
fragment of a lichen from which we attempted to amplify and sequence
green algal ITS2 rDNA. We attempted to amplify algal DNA from four
sections of each lichen in order to quantity algal diversity within a
single lichen. If sequencing was successful then the DNA sequence is
given along with a code identifying unique haplotype (GenotypeID). The
TaxonID column indicates haplotypes that matched those reported in <a
href="http://dw.doi.org/10.3732/ajb.0900276">Werth and Sork
(2010)</a>.</li>
</ol>
<p>The R script explores algal diversity among lichens and trees.
Working in pairs, execute each line of code and then add a comment
(using <code>#</code>) above each line of code with a description of
what the code does.</p>
<p>When you have finished commenting the code, try to complete the
following challenge:</p>
<blockquote>
<p><strong>Challenge:</strong></p>
<p>How could you modify the code that creates the
<code>lichenXgeno</code> table so that it displays the fraction of
successful sequences belonging to each GenotypeID for each lichen.</p>
</blockquote>
</div>
<div id="plotting-data-with-ggplot2" class="section level3">
<h3>Plotting data with ggplot2</h3>
<p>Working with your partner, create the following plots using
<code>ggplot2</code>:</p>
<ol style="list-style-type: decimal">
<li>A plot that maps the locations where each lichen was collected and
colors the point by the Chao1 richness estimator.</li>
<li>Modify the above plot so that three plots are shown, one for each
tree species.</li>
</ol>
</div>

<hr>
<p>Introduction to Computing in R with NEON</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
